Client/Frontend/Settings/ClearPrivateDataTableViewController.swift

```diff
@@ -106,28 +108,40 @@ class ClearPrivateDataTableViewController: UITableViewController {
     override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
         guard indexPath.section == SectionButton else { return }
 
-        let toggles = self.toggles
-        clearables
-            .enumerate()
-            .flatMap { (i, pair) in
-                guard toggles[i] else {
-                    return nil
+        func clearPrivateData() {
+            let toggles = self.toggles
+            self.clearables
+                .enumerate()
+                .flatMap { (i, pair) in
+                    guard toggles[i] else {
+                        return nil
+                    }
+                    log.debug("Clearing \(pair.clearable).")
+                    return pair.clearable.clear()
                 }
-                log.debug("Clearing \(pair.clearable).")
-                return pair.clearable.clear()
-            }
-            .allSucceed()
-            .upon { result in
-                assert(result.isSuccess, "Private data cleared successfully")
+                .allSucceed()
+                .upon { result in
+                    assert(result.isSuccess, "Private data cleared successfully")
 
-                self.profile.prefs.setObject(self.toggles, forKey: TogglesPrefKey)
+                    self.profile.prefs.setObject(self.toggles, forKey: TogglesPrefKey)
 
-                dispatch_async(dispatch_get_main_queue()) {
-                    // Disable the Clear Private Data button after it's clicked.
-                    self.clearButtonEnabled = false
-                    self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
-                }
+                    dispatch_async(dispatch_get_main_queue()) {
+                        // Disable the Clear Private Data button after it's clicked.
+                        self.clearButtonEnabled = false
+                        self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
+                    }
+            }
+        }
+
+        // If history is being cleared and the user is logged in then we ask for confirmation. Otherwise just do it.
+        if self.toggles[HistoryClearableIndex] && profile.hasAccount() {
```

Comment:
This isn't quite enough. A user who's not syncing history is going to be really alarmed if we tell them that their history will be cleared on their desktop!

Please add a note to that effect and file a follow-up bug, and I'll give some guidance in the bug about how to figure that out.